### ISSUE TITLE
KeyError: 'example_value' while trying to trace TransformerEncoderLayer

### ISSUE THREAD

**Issue by bertmaher** [facebookresearch/torchdynamo#175]
See this gist for a repro: https://gist.github.com/bertmaher/f581734cf80b967592ccdff6d513774b

The model here is more or less a copypaste of nn.TransformerEncoderLayer, but applying `torchdynamo.optimize` yields the following stack trace:
```
ERROR FROM offset=18 filename encoder.py 42 KeyError
ERROR FROM offset=20 filename encoder.py 36 KeyError
========== TorchDynamo Stack Trace ==========
Traceback (most recent call last):
  File "/fsx/users/bertrand/torchdynamo/torchdynamo/convert_frame.py", line 163, in _convert_frame_assert
    code = transform_code_object(frame.f_code, transform)
  File "/fsx/users/bertrand/torchdynamo/torchdynamo/bytecode_transformation.py", line 284, in transform_code_object
    transformations(instructions, code_options)
  File "/fsx/users/bertrand/torchdynamo/torchdynamo/convert_frame.py", line 139, in transform
    tracer.run()
  File "/fsx/users/bertrand/torchdynamo/torchdynamo/symbolic_convert.py", line 278, in run
    and self.step()
  File "/fsx/users/bertrand/torchdynamo/torchdynamo/symbolic_convert.py", line 256, in step
    getattr(self, inst.opname)(inst)
  File "/fsx/users/bertrand/torchdynamo/torchdynamo/symbolic_convert.py", line 142, in wrapper
    return inner_fn(self, inst)
  File "/fsx/users/bertrand/torchdynamo/torchdynamo/symbolic_convert.py", line 556, in CALL_FUNCTION
    self.call_function(fn, args, {})
  File "/fsx/users/bertrand/torchdynamo/torchdynamo/symbolic_convert.py", line 195, in call_function
    self.push(fn.call_function(self, args, kwargs))
  File "/fsx/users/bertrand/torchdynamo/torchdynamo/variables/functions.py", line 182, in call_function
    return super().call_function(tx, args, kwargs)
  File "/fsx/users/bertrand/torchdynamo/torchdynamo/variables/functions.py", line 64, in call_function
    return tx.inline_user_function_return(
  File "/fsx/users/bertrand/torchdynamo/torchdynamo/symbolic_convert.py", line 227, in inline_user_function_return
    result = InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
  File "/fsx/users/bertrand/torchdynamo/torchdynamo/symbolic_convert.py", line 1150, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
  File "/fsx/users/bertrand/torchdynamo/torchdynamo/symbolic_convert.py", line 1192, in inline_call_
    tracer.run()
  File "/fsx/users/bertrand/torchdynamo/torchdynamo/symbolic_convert.py", line 278, in run
    and self.step()
  File "/fsx/users/bertrand/torchdynamo/torchdynamo/symbolic_convert.py", line 256, in step
    getattr(self, inst.opname)(inst)
  File "/fsx/users/bertrand/torchdynamo/torchdynamo/symbolic_convert.py", line 142, in wrapper
    return inner_fn(self, inst)
  File "/fsx/users/bertrand/torchdynamo/torchdynamo/symbolic_convert.py", line 605, in CALL_FUNCTION_KW
    self.call_function(fn, args, kwargs)
  File "/fsx/users/bertrand/torchdynamo/torchdynamo/symbolic_convert.py", line 195, in call_function
    self.push(fn.call_function(self, args, kwargs))
  File "/fsx/users/bertrand/torchdynamo/torchdynamo/variables/nn_module.py", line 172, in call_function
    return variables.TensorVariable.create(
  File "/fsx/users/bertrand/torchdynamo/torchdynamo/variables/tensor.py", line 122, in create
    TensorVariable.create(
  File "/fsx/users/bertrand/torchdynamo/torchdynamo/variables/tensor.py", line 75, in create
    args, kwargs = cls.propagate_args_kwargs(proxy.node)
  File "/fsx/users/bertrand/torchdynamo/torchdynamo/variables/tensor.py", line 60, in propagate_args_kwargs
    return torch.fx.node.map_arg((node.args, node.kwargs), visit)
  File "/fsx/users/bertrand/conda/envs/triton3/lib/python3.8/site-packages/torch/fx/node.py", line 598, in map_arg
    return map_aggregate(a, lambda x: fn(x) if isinstance(x, Node) else x)
  File "/fsx/users/bertrand/conda/envs/triton3/lib/python3.8/site-packages/torch/fx/node.py", line 606, in map_aggregate
    t = tuple(map_aggregate(elem, fn) for elem in a)
  File "/fsx/users/bertrand/conda/envs/triton3/lib/python3.8/site-packages/torch/fx/node.py", line 606, in <genexpr>
    t = tuple(map_aggregate(elem, fn) for elem in a)
  File "/fsx/users/bertrand/conda/envs/triton3/lib/python3.8/site-packages/torch/fx/node.py", line 606, in map_aggregate
    t = tuple(map_aggregate(elem, fn) for elem in a)
  File "/fsx/users/bertrand/conda/envs/triton3/lib/python3.8/site-packages/torch/fx/node.py", line 606, in <genexpr>
    t = tuple(map_aggregate(elem, fn) for elem in a)
  File "/fsx/users/bertrand/conda/envs/triton3/lib/python3.8/site-packages/torch/fx/node.py", line 616, in map_aggregate
    return fn(a)
  File "/fsx/users/bertrand/conda/envs/triton3/lib/python3.8/site-packages/torch/fx/node.py", line 598, in <lambda>
    return map_aggregate(a, lambda x: fn(x) if isinstance(x, Node) else x)
  File "/fsx/users/bertrand/torchdynamo/torchdynamo/variables/tensor.py", line 58, in visit
    return n.meta["example_value"]
KeyError: 'example_value'
```

It seems to have something to do with the MultiheadAttention module attr, but I got only far enough into debugging to realize that it was better to ask y'all.

cc @anj-s

**Comment by anijain2305** [comment]
I can take a look at this tomorrow. I am dealing with these type of issues at many places.

**Comment by jansel** [comment]
@anijain2305 I didn't see your comment, sorry!  I fixed it in #176.

The issue was `nn.MultiheadAttention()` returned `(torch.Tensor(), None)` and that `None` confused `TensorVariable.create()` because when it is given `example_value=None` as a kwarg it assumed it needed to compute it.

**Comment by anijain2305** [comment]
Oh no worries at all :)
Thanks for the quick fix.
